llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp

```diff
@@ -3112,6 +3112,26 @@ Instruction *InstCombinerImpl::foldSelectOfBools(SelectInst &SI) {
     return SelectInst::Create(FalseVal, One, AndV);
   }
 
+  // select (~b & a), a, b -> or a, b
+  // only for scalar types
+  if (match(CondVal, m_c_And(m_Not(m_Specific(FalseVal)), m_Specific(TrueVal))) &&
+      TrueVal->getType()->isIntegerTy(1) &&
+      FalseVal->getType()->isIntegerTy(1) &&
+      CondVal->getType()->isIntegerTy(1) &&
```

Comment:
These scalar type checks are not necessary. When given vector operands, the `select` instruction will perform the selection element by element.